lib/services/history/history_manager.ts

```diff
@@ -5,26 +5,33 @@ import { SNItem } from '@Models/core/item';
 import { ContentType } from '@Models/content_types';
 import { PureService } from '@Lib/services/pure_service';
 import { HistorySession } from '@Services/history/history_session';
+import { ServerHistoryRevision } from '@Lib/services/history/server_history_revision';
 import { PayloadSource } from '@Payloads/sources';
 import { StorageKey } from '@Lib/storage_keys';
 import { isNullOrUndefined, concatArrays } from '@Lib/utils';
+import { SNApiService } from '@Services/api/api_service';
+import { PurePayload } from '@Lib/protocol/payloads';
+import { SNProtocolService } from '../protocol_service';
 
 const PERSIST_TIMEOUT = 2000;
 
 /**
- * The history manager is presently responsible for transient 'session history',
+ * The history manager is responsible for transient 'session history',
  * which include keeping track of changes made in the current application session.
  * These change logs (unless otherwise configured) are ephemeral and do not persist
  * past application restart.
- * In the future the history manager will also be responsible for remote server history.
+ * History manager is also responsible for remote server history.
  */
 export class SNHistoryManager extends PureService {
 
   private itemManager?: ItemManager
   private storageService?: SNStorageService
+  private apiService?: SNApiService
+  private protocolService?: SNProtocolService
   private contentTypes: ContentType[] = []
   private timeout: any
   private historySession?: HistorySession
+  private historyServer?: ServerHistoryRevision
```

Comment:
The naming for historySession makes sense because it is quite literally a session—a transient structure holding data. But with `historyServer`, that naming doesn't make sense. This should be named something else. Also, I don't even see why you need a property here? When a client accesses server history, there should be no state saved—it's all a matter of API requests, and we simply relay results to the client. The client then displays those results. When the client dismisses the history menu, the history data vanishes.